### DISCUSSION TITLE
Which fields does database elasticsearch use to determine whether a service instance is offline?

### DISCUSSION THREAD

**Discussion by mitht** [apache/skywalking#11642] [Q&A] [answered]
apm version 9.3.0
agent version 8.14.0
I want to know whether the service instance is online through a database field

**Comment by weixiang1862** [comment, marked as answer]
instance_traffic last_ping field, it records agent last heartbeat time.

**Reply by mitht** [reply]
What is the threshold?

**Reply by weixiang1862** [reply]
depend on agent config: `collector.heartbeat_period=${SW_AGENT_COLLECTOR_HEARTBEAT_PERIOD:30}`

**Reply by mitht** [reply]
Thank you so much for your help!
